src/main/php/cfg/verb/verb.php

```diff
@@ -756,140 +758,144 @@ private function save_field_do(sql_db $db_con, $log): string
             if ($this->can_change()) {
                 $db_con->set_class(verb::class);
                 if (!$db_con->update_old($this->id(), $log->field(), $new_value)) {
-                    $result .= 'updating ' . $log->field() . ' to ' . $new_value . ' for verb ' . $this->dsp_id() . ' failed';
+                    $usr_msg->add_id_with_vars(msg_id::VERB_UPDATE_FAILED, [
+                        msg_id::VAR_NAME => $log->field(),
+                        msg_id::VAR_VALUE => $new_value,
+                        msg_id::VAR_ID => $this->dsp_id()
+                    ]);
                 }
 
             } else {
                 // TODO: create a new verb and request to delete the old
                 log_warning('verb->save_field_do creating of a new verb not yet coded');
             }
         }
-        return $result;
+        return $usr_msg;
     }
 
-    private function save_field_code_id(sql_db $db_con, $db_rec): string
+    private function save_field_code_id(sql_db $db_con, $db_rec): user_message
     {
-        $result = '';
+        $usr_msg = null;
```

Comment:
Why not return an empty user_message object instead of null? Yes both options have its pros and cons, but until now I have usually used an empty user_message object to reduce the is null check in the calling function